packages/react-components/react-charts-preview/library/src/components/CommonComponents/Popover.tsx

```diff
@@ -0,0 +1,247 @@
+import { PopoverComponentProps } from './Popover.types';
+
+const PopoverComponent: React.FunctionComponent<PopoverComponentProps> = React.forwardRef<
+  HTMLDivElement,
+  PopoverComponentProps
+>((props, forwardedRef) => {
+  const virtualElement: PositioningVirtualElement = {
+    getBoundingClientRect: () => ({
+      top: props.clickPosition!.y,
+      left: props.clickPosition!.x,
+      right: props.clickPosition!.x,
+      bottom: props.clickPosition!.y,
+      x: props.clickPosition!.x,
+      y: props.clickPosition!.y,
+      width: 0,
+      height: 0,
+    }),
+  };
+  props = { ...props, ...props.customCallout?.customCalloutProps };
+  const classes = usePopoverStyles_unstable(props);
+  const Legend = props.xCalloutValue ? props.xCalloutValue : props.legend;
+  const YValue = props.yCalloutValue ? props.yCalloutValue : props.YValue;
+  return (
+    <div id={useId('callout')}>
+      <Popover
+        positioning={{ target: virtualElement, autoSize: 'always', offset: 20, coverTarget: false }}
+        open={props.isPopoverOpen}
+        inline
+      >
+        <PopoverSurface>
+          {/** Given custom callout, then it will render */}
+          {props.customCallout && props.customCallout.customizedCallout && props.customCallout.customizedCallout}
+          {/** single x point its corresponding y points of all the bars/lines in chart will render in callout */}
+          {(!props.customCallout || !props.customCallout.customizedCallout) &&
+            props.isCalloutForStack &&
+            _multiValueCallout()}
+          {/** single x point its corresponding y point of single line/bar in the chart will render in callout */}
+          {(!props.customCallout || !props.customCallout.customizedCallout) && !props.isCalloutForStack && (
+            <div className={classes.calloutContentRoot}>
+              <div className={classes.calloutDateTimeContainer}>
+                <div className={classes.calloutContentX}>{props.XValue} </div>
+                {/*TO DO  if we add time for callout then will use this */}
+                {/* <div className={classNames.calloutContentX}>07:00am</div> */}
+              </div>
+              <div
+                className={classes.calloutInfoContainer}
+                style={{
+                  ...(props.ratio && {
+                    display: 'flex',
+                    alignItems: 'flex-end',
+                  }),
+                  borderLeft: `4px solid ${props.color}`,
+                }}
+              >
+                <div className={classes.calloutBlockContainer}>
+                  <div className={classes.calloutlegendText}>{convertToLocaleString(Legend, props.culture)}</div>
+                  <div
+                    className={classes.calloutContentY}
+                    style={{ color: props.color ? props.color : tokens.colorNeutralForeground1 }}
+                  >
+                    {convertToLocaleString(YValue, props.culture)}
```

Comment:
Another case where the call can be made in script rather than TSX.

Reply:
These logic are directly ported over from v8. We will address these in upcoming iterations to improve the component performance.
Tracking this comment here https://github.com/microsoft/fluentui/issues/33042